src/NuGet.Clients/NuGet.CommandLine/Commands/DeprecatedCommandAttribute.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+
+namespace NuGet.CommandLine
+{
+    /// <summary>
+    /// Attribute for marking nuget.exe commands as deprecated.
+    ///
+    /// Using this attribute will in ICommand--based classes, it will show a warning message each
+    /// command invocation and in each help command invocation.
+    ///
+    /// You need to provide an alternative command to use this attribute
+    ///
+    /// 
+    /// <see cref="ICommand"/>
+    /// <see cref="Command"/>
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false, Inherited = false)]
+    public sealed class DeprecatedCommandAttribute : Attribute
+    {
+        public Type AlternativeCommand { get; set; }
+
+        public DeprecatedCommandAttribute(Type alternativeCommand)
+        {
+            if (alternativeCommand == null || !typeof(ICommand).IsAssignableFrom(alternativeCommand))
```

Comment:
I don't think the annotation needs to have a mandatory alternative command. 

Sometimes commands and their functionality are deprecated completely.

Reply:
Agree. Now there's an option for not providing an alternative command